### DISCUSSION TITLE
Initialize vtk DataSet from clientfile

### DISCUSSION THREAD

**Discussion by bigfooted** [Kitware/trame#308] [General] [answered]
Hi! I am reading a .vtm file from disk and then visualizing this. I used to do:
```
reader = vtk.vtkXMLMultiBlockDataReader()
reader.SetFileName(filename)
reader.Update()
...
mb = reader.GetOutput()
```

But now I want to switch to a server setup, where the file is uploaded, so I am using the setup from the howdoi/00_upload example and do:
`file = ClientFile(file_upload)`

and now I am confused. Is there a way to directly update the variable mb with the data inside file?

**Comment by jourdain** [comment, marked as answer]
So the ClientFile is actually only a set of buffers in memory. You will need to dump it to disk and point your VTK reader to it so it can produce a new multiblock. Also if I'm not mistaken, a vtm is a meta file which require other files on the side. So uploading a vtm in itself won't be enough unless you now where is was located and the server already has access to the other files.

HTH

**Reply by bigfooted** [reply]
OK, clear, so I basically just upload the file(s) to the server and then use the reader as I did before, using the file_binary_content:

```
with open(file_name, "wb") as binary_file:
    binary_file.write(file_binary_content)
```

**Reply by bigfooted** [reply]
By the way, is it possible to upload a folder + everything inside it? I looked at the vuetify v-file-input options but could only find the option webkitdirectory, and that did not seem to do anything when I set it to True in vuetify.VFileInput

**Reply by jourdain** [reply]
If it is a true upload (meaning server is not collocated with the client [localhost]), there is no way. This is a limitation of the web API. Chrome has some experimental support but I don't think it went thru to the std and it won't.

If not, you can use something like an embedded [file browser](https://github.com/Kitware/trame-pvui/blob/master/trame_pvui/widgets/filebrowser.py) (I think I posted a picture of it in another discussion but can't find it) so you can get the path and don't move the data at all.

**Reply by bigfooted** [reply]
Thanks! I think you mean this one:
https://github.com/Kitware/trame/discussions/294

The .vtm file is basically a file that contains a list of files. Maybe it is possible to extract these filenames and parse it to the uploader?

**Reply by bigfooted** [reply]
or else I can just zip the vtm files, upload that and unzip it on the server

**Reply by jourdain** [reply]
The zip is probably the only approach

**Reply by jourdain** [reply]
unless you can figure out another format within a single file.
